.github/workflows/publish.yml

```diff
@@ -29,4 +33,11 @@ jobs:
           publish: pnpm run changeset:publish
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          NPM_TOKEN: ${{ secrets.WORKLEAP_NPM_TOKEN }}
\ No newline at end of file
+          NPM_TOKEN: ${{ secrets.WORKLEAP_NPM_TOKEN }}
+      
+      - name: LinearB
+        if: ${{ steps.changesets.outputs.published == 'true' }}
+        uses: ./.github/actions/linearb-main
+        with:
+          environment: release
+          timestamp: ${{ steps.linearb-pre.outputs.timestamp }}
```

Comment:
I don't think you need to bother with the timestamp. Once the PR is merged, you can trigger the linearb steps. The start of the pipeline or this step should almost have the same timestamp. 
I'm not familiar with this flow, so maybe I'm wrong.

So, I think you can reuse the existing shared steps and call it here.

Reply:
No, I'm pretty sure I'm the one who did the wrong thing by assuming the timestamp had to be at the beginning of the workflow. I should be able to reuse the shared workflow.